saleor/order/utils.py

```diff
@@ -195,19 +203,16 @@ def update_order_status(order: Order):
             total_quantity,
             quantity_fulfilled,
             quantity_returned,
+            quantity_awaiting_approval,
         ) = _calculate_quantity_including_returns(locked_order)
 
-        # check if order contains any fulfillments that awaiting approval
-        awaiting_approval = locked_order.fulfillments.filter(
-            status=FulfillmentStatus.WAITING_FOR_APPROVAL
-        ).exists()
-
         status = determine_order_status(
             order,
             total_quantity,
             quantity_fulfilled,
             quantity_returned,
-            awaiting_approval,
+            quantity_awaiting_approval,
+            confirm,
         )
```

Comment:
Maybe instead of checking `confirm` flag and `order.is_unconfirmed()` condition in  `determine_order_status` function, you can quit the method earlier. I mean:

```suggestion
        if not confirm and order.is_unconfirmed():
            return
     
        status = determine_order_status(
            order,
            total_quantity,
            quantity_fulfilled,
            quantity_returned,
            quantity_awaiting_approval,
        )
```
As in such case we do not want to change the status at all, right?

Actually, we can even extend the condtiton with `total_quantity == 0` check.
```suggestion
        all_products_replaced = total_quantity == 0
        skip_update_for_unconfirmed = not confirm and order.is_unconfirmed()
        if skip_update_for_unconfirmed or all_products_replaced:
            return
     
        status = determine_order_status(
            order,
            total_quantity,
            quantity_fulfilled,
            quantity_returned,
            quantity_awaiting_approval,
        )
```

Reply:
As you prefer. I actually did something similar initially, but then decided against it to keep all the conditions in determine_order_status function. My reasoning was that in such case, it's easier to understand what is happening and why. But if you prefer this way, I don't object :)

Reply:
@IKarbowiak When I do this change, maybe I will incorporate „automatically_confirm_all_new_orders” check in „determine_order_status”? Then we don’t have extra attribute and we can remove copy pasted check from 3 different ifs